### PR TITLE
feat: Adding Buttons to download selected systems or all systems

### LINKED ISSUE #69: ui provides a link to download scores as excel file

### PR DESCRIPTION
**Pull request by ATNoblis**
### Summary

Adding two buttons that allow users to download the selected systems or all systems they are assigned.

closes #69 

### Added

None

### Changes

FismaTable.jsx - adding checkboxes for users to selected. Create custom footer such that it added the two buttons. 

Snackbar.jsx - created a more generic snackbar to pass different severities and text.

UserTable.jsx - handle changes of snackbar to be used in user table

### Test

Try the buttons at the bottom of dashboard.
- Download icon is to download systems selected

### CODE REVIEW

File: src/views/FismaTable/FismaTable.tsx [hunk 1]
```diff
@@ -1,17 +1,135 @@
...
+        })
+        .then((response) => {
+          if (response.status !== 200) {
+            console.log('Error saving systems')
+          } else {
+            const contentType = response.headers['content-type']
+            const data = new Blob([response.data], { type: contentType })
+            const tempLink = document.createElement('a')
+            tempLink.href = window.URL.createObjectURL(data)
+            tempLink.setAttribute('download', 'ZTMF_FY2023Q4.xlsx')
```
Comment (talentedmrjones) [inline src/views/FismaTable/FismaTable.tsx:63]: is this a hard coded filename?

**Review by talentedmrjones — CHANGES_REQUESTED** [review]

**Review by talentedmrjones — CHANGES_REQUESTED** [review]
Lets just have 1 icon and a way to check/uncheck all systems

File: src/views/FismaTable/FismaTable.tsx [hunk 2]
```diff
@@ -1,17 +1,123 @@
...
+    } else {
+      let exportUrl = '/datacalls/2/export'
+      let exportFileName = ''
+      if (
+        props.selectedRows &&
+        props.fismaSystems &&
+        props.selectedRows.length < props.fismaSystems.length
+      ) {
+        exportUrl += '?'
+        exportFileName = 'Selected_Systems_Export.xlsx'
```
Comment (talentedmrjones) [inline src/views/FismaTable/FismaTable.tsx:50]: The backend already provides the generated filename in the response headers. Remove this so the browser uses what the backend provides

**Review by talentedmrjones — CHANGES_REQUESTED** [review]

File: src/views/FismaTable/FismaTable.tsx (hunk 2 above)
Comment (ATNoblis) [reply src/views/FismaTable/FismaTable.tsx]: https://github.com/CMS-Enterprise/ztmf-ui/blob/c010edd8990bf658a21dbd486c9d0d321ddc0d54/src/views/FismaTable/FismaTable.tsx#L68-L69

**Review by talentedmrjones — APPROVED** [review]

**Pull request merged**